actors/builtin/miner/miner_actor.go

```diff
@@ -652,7 +698,9 @@ type TerminateSectorsParams struct {
 func (a Actor) TerminateSectors(rt Runtime, params *TerminateSectorsParams) *adt.EmptyValue {
 	var st State
 	rt.State().Readonly(&st)
-	rt.ValidateImmediateCallerIs(st.Info.Worker)
+	info, err := st.GetInfo(adt.AsStore(rt))
```

Comment:
There seems to be a number of places where we access info solely to validate the caller is worker. Consider extracting a function.

Reply:
Seeing more cases, maybe just a function to get the info obj and do the error check.

Reply:
Such a function would need the `rt` as an argument, which is against the general policy. It's a decent example of a pragmatic exception, though, so 👍 .

I'd prefer caller validation remains explicitly in the top-level actor method wherever possible.

This should be a free function, rather than method on state. `info := getMinerInfo(rt, &st)`